### PR TITLE
16 entities create initial entities

### PR DESCRIPTION
**Pull request by AlekseyPanas**

### CODE REVIEW

**Review by cotopaxi64 — DISMISSED** [review]
Looks good, great documentation. We can merge this for now, but afterwards we need to write the tests.

File: src/main/java/entities/LobbyManager.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+        // Below this, we assume that l is in pool and game is running. We assume a proper
+        // lock architecture that ensures that this assumption cannot change from another thread
+
+        // Removes the link from pool
+        playerPool.remove(l);
+
+        // Passes game to the corresponding listener
+        l.playerPoolListener.onJoinGamePlayer(game);
+
+
```
Comment (bac423) [inline src/main/java/entities/LobbyManager.java:127]: remove empty line

**Review by Serdiuk-Andrii — APPROVED** [review]

File: src/main/java/entities/PlayerFactory.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package entities;
+
+import exceptions.IdInUseException;
+
+import java.util.ArrayList;
+
```
Comment (bac423) [inline src/main/java/entities/PlayerFactory.java:6]: Add small doc

File: src/main/java/entities/games/GameFactory.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+public interface GameFactory {
+    /**
+     * Given some arguments, create the appropriate game instance
+     * @param settings A map of strings to integer settings
+     * @param initialPlayers A list of initial players
+     * @return the created game instance
+     */
+    Game createGame(Map<String, Integer> settings, Collection<Player> initialPlayers);
+}
```
Comment (bac423) [inline src/main/java/entities/games/GameFactory.java:16]: add new line at the end of the file

File: src/main/java/entities/games/GameFactory.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package entities.games;
+
+import entities.Player;
+import java.util.Collection;
+import java.util.Map;
+
+
```
Comment (bac423) [inline src/main/java/entities/games/GameFactory.java:7]: add small doc

File: src/main/java/entities/games/GameFactoryRegular.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package entities.games;
+
+import entities.Player;
+
+
+import java.util.*;
+
```
Comment (bac423) [inline src/main/java/entities/games/GameFactoryRegular.java:7]: add small doc

**Review by bac423 — APPROVED** [review]
Several docs missing, otherwise fine

File: src/main/java/entities/Story.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import exceptions.InvalidWordException;
+
+import java.util.List;
+
+/**
+ * Story that is made up of words
+ */
+public class Story {
+
+    List<Word> words;
```
Comment (AlekseyPanas) [inline src/main/java/entities/Story.java:12]: Make Private

**Review by tostistos — APPROVED** [review]
Looks good, any issues I found intially were fixed

**Review by Serdiuk-Andrii — APPROVED** [review]

**Pull request merged**